docs/developer/extending/apps/architecture/app-permissions.mdx

```diff
@@ -0,0 +1,174 @@
+
+In your React component, you can use the `useAppBridge` hook from [@saleor/app-sdk](developer/extending/apps/developing-apps/app-sdk/overview.mdx) to check what permissions the user has.
+
+We will compare the results with the required permission and render the table accordingly:
+
+```jsx
+import { useAppBridge } from "@saleor/app-sdk";
+import { OrdersTable } from "./OrdersTable";
+
+const Component = () => {
+  const { appBridgeState } = useAppBridge();
+
+  const hasPermission =
+    appBridgeState.user.permissions.includes("MANAGE_ORDERS");
+
+  return hasPermission ? (
+    <OrdersTable />
+  ) : (
+    <p>You don't have the required permissions</p>
+  );
+};
+```
+
+Once you have confirmed that the user has the required permissions, you can execute the `orders` query in your `OrdersTable` component:
+
+:::info
+
+In the example below, you can see passing the `appBridgeState.token` right to the `Authorization` header in the query. This is for presentational purposes only. You should rather create a client with the token and pass it to the provider of your GraphQL client library.
+
+See: [App Template](https://github.com/saleor/saleor-app-template/blob/main/src/providers/GraphQLProvider.tsx) for reference.
+
+:::
+
+```jsx
+import { useAppBridge } from "@saleor/app-sdk";
+import { useQuery } from "grapghql-client-library"; // made up library
+import { ORDERS_QUERY } from "./queries";
+
+export const OrdersTable = () => {
+  const { appBridgeState } = useAppBridge();
+  const { data } = useQuery(ORDERS_QUERY, {
+    context: {
+      headers: {
+        Authorization: `Bearer ${appBridgeState.token}`, // this is the user token
+      },
+    },
+  });
+
+  // ...
+};
+```
+
+Summing up, the user scope applies to the client-side of our app. In the case of the Orders App, it means that as long as the user has the `MANAGE_ORDERS` permission, they will see the orders table.
+
+### App scope
+
+When you want to execute a query that requires permission that the user may not have, you should do it on the server-side.
+
+The app scope applies to the server-side of our app. It means that as long as the app has the `MANAGE_ORDERS` permission, it can execute the `orders` query, regardless of the user's permissions.
+
+In our case, you would execute the `orders` query on the server-side and pass the result to the client-side. In Next.js, it would mean creating an API route and executing the query there:
+
+```jsx
+// pages/api/orders.js
+import { SALEOR_API_URL_HEADER } from "@saleor/app-sdk/const";
+import { ordersQuery } from "./queries";
+import { createClient } from "graphql-client-library"; // made up library
+
+export default async function handler(req, res) {
+  const saleorApiUrl = req.headers[SALEOR_API_URL_HEADER]; // assuming it's set
```

Comment:
Lets add note with link to the nextjs validation helpers we have in the SDK - those can be used to check user token/permissions.

Reply:
https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-apps/app-sdk/protected-handlers